### DISCUSSION TITLE
Have no idea how to make a link rotate around its parent joint

### DISCUSSION THREAD

**Discussion by zhuoqun-chen** [dfki-ric/phobos#308] [Q&A] [answered]
Dear developers,

I use phobos to load a quadrupedal robot urdf into my scene and I want to use python to set the joint angles for each frame to rotate links. But It seems that I can only directly set rotation `XYZ Euler` to the armature, but the animation seems to be incorrect:

![image](https://github.com/dfki-ric/phobos/assets/73287160/6ca5dc96-2d05-4d34-828c-c43501f0f4f7)

You can see in this picture that the `calf` link rotates exceeding the maximum limitation around the joint (bone) during the animation, however, the value used is within the safe range. This may imply, I guess, that only rotate the armature is somehow incorrect.

So I want to rotate bones in `edit mode` or `pose mode` to see if rotating joints (bones) will be correct. But I can't even rotate bones directly in blender's GUI.

![image](https://github.com/dfki-ric/phobos/assets/73287160/bbadfb83-1929-4572-8f2b-ce0584e6668a)

Could you give me some help or advice? Thank you very much!

**Comment by hwiedPro** [comment, marked as answer]
Hi @zhuoqun-chen,
can you provide some information on how you defined the joint? Did you do it manually, loaded it from URDF etc...?

**Reply by zhuoqun-chen** [reply]
Hi hwiedPro,

Thank you for the quick reply! And I solved the problem already. 

I found that I have to use `bpy` to control the `pose bone` Euler Angle directly instead of the `armature` Euler Angle. This may be because `pose bone` corresponds to actual `joint` so it respect the maximum and minimum limitations (armature doesn't).

**Reply by hwiedPro** [reply]
I'm happy that you found a way and thanks for sharing your solution!
